pkg/crc/ssh/client.go

```diff
@@ -84,26 +87,49 @@ func (client *NativeClient) session() (*ssh.Session, error) {
 		if err != nil {
 			return nil, err
 		}
+
+		if err := client.conn.SetDeadline(time.Now().Add(timeout)); err != nil {
+			return nil, err
+		}
+		defer func() {
+			_ = client.conn.SetDeadline(time.Time{})
+		}()
+
```

Comment:
This does not need to be duplicated here as you are doing it after the end of the block anyway? Or can this particular deadline be triggered as part of `NewClientConn`?
Could we only call  `SetDeadline` once when creating `client.sshClient`? If a different timeout is needed, `session` can be called again with the new timeout value?

Reply:
This code is confusing me, it's written as if we could call `session()` multiple times with different timeout values, and it will create the low-level connection if needed, or reuse an existing one, and set the timeout as expected in all cases. But the only user of this method just does `session := session(); session.Close()`, so we don't have a need for something so sophisticated?